### ISSUE TITLE
Automatically use LibYAML bindings if they are available (or easily enable globally)

### ISSUE THREAD

**Issue by psphicas** [yaml/pyyaml#437]
It would be nice to have fast LibYAML bindings automatically enabled if they are available, or, to have an easy way to turn them on globally, without modifying existing code that already uses PyYAML, and that can work even when importing other packages that themselves use PyYAML.

The suggestion in the documentation
```python
from yaml import load, dump
try:
    from yaml import CLoader as Loader, CDumper as Dumper
except ImportError:
    from yaml import Loader, Dumper
# ...
data = load(stream, Loader=Loader)
# ...
output = dump(data, Dumper=Dumper)
```
only works for code that is fully controlled. Also, it is inconvenient to retrofit large projects that may import yaml in more than one place, use load, dump, add_representer, etc. without explicitly specifying a Loader or Dumper, or use the "sugar" methods.

This seems work, but maybe there is a better way?
#436

**Comment by bryant1410** [comment]
What about having a `yaml.auto` import that automatically selects the `Loader`, `Dumper`, etc? Like [`tqdm.auto` API](https://tqdm.github.io/docs/shortcuts/#tqdmauto). Which could eventually become the default in a major version change (v7?).

**Comment by nitzmahone** [comment]
Yeah, this has been a long-standing need, but is somewhat complicated in the current codebase by the way the various Loaders and Dumpers are technically unrelated concrete classes composed entirely of mixins. I have a working sample of something similar in #700 - there's still some bikeshedding on the name, but the overall concept seems workable (since the new API lets you explicitly opt in to the features/tags you want independent of the underlying Loader/Dumper impl).
